pkg/operator/sync_openshiftcontrollermanager_v311_00.go

```diff
@@ -189,6 +195,38 @@ func manageOpenShiftControllerManagerConfigMap_v311_00_to_latest(kubeClient kube
 	return resourceapply.ApplyConfigMap(client, recorder, requiredConfigMap)
 }
 
+func manageOpenShiftRegistryCAConfigMap_v311_00_to_latest(kubeClient kubernetes.Interface, client coreclientv1.ConfigMapsGetter, recorder events.Recorder) (bool, error) {
+	configMap := resourceread.ReadConfigMapV1OrDie(v311_00_assets.MustAsset("v3.11.0/openshift-controller-manager/openshift-registry-ca-cm.yaml"))
+	existing, err := client.ConfigMaps(util.TargetNamespace).Get("openshift-registry-ca", metav1.GetOptions{})
+	// Ensure we create the ConfigMap for the registry CA, and that it has the right annotations
+	// Lifted from library-go for the most part
+	if apierrors.IsNotFound(err) {
+		_, err := client.ConfigMaps(util.TargetNamespace).Create(configMap)
+		if err != nil {
+			recorder.Eventf("ConfigMapCreateFailed", "Failed to create %s%s/%s%s: %v", "configmap", "", "openshift-registry-ca", "-n openshift-controller-manager", err)
+			return true, err
+		}
+		recorder.Eventf("ConfigMapCreated", "Created %s%s/%s%s because it was missing", "configmap", "", "openshift-registry-ca", "-n openshift-controller-manager")
+		return true, nil
+	}
+
+	// Ensure the openshift-registry-ca ConfigMap has the service.beta.openshift.io/inject-cabundle annotation
+	// Otherwise ignore the contents of the ConfigMap
+	modified := resourcemerge.BoolPtr(false)
+	existingCopy := existing.DeepCopy()
+	resourcemerge.EnsureObjectMeta(modified, &existingCopy.ObjectMeta, configMap.ObjectMeta)
```

Comment:
how are you checking whether the service-ca value within the configmap has changed?

in our other configmaps we store a hash of the keys/values so we know if something has changed that needs to force a rollout of the configmap consumer....

Reply:
you should be able to just add that in manageOpenShiftControllerManagerConfigMap.  In fact isn't that already watching a client-ca configmap?  (maybe for a differnet ca)

Reply:
yes, we are watching the client-ca ConfigMap and the serving-cert secret.
We'll still need this function to create the `openshift-service-ca` ConfigMap and ensure that it has the `inject-cabundle` annotation.

Reply:
oh. you do not need to watch the openshift-service-ca configmap from here because the controller-manager itself does not care if that configmap changes.  only the buildcontroller logic inside the controllermanager, which already watches the configmap itself.

Reply:
@bparees renamed ConfigMap to `openshift-service-ca` - ptal